### ISSUE TITLE
Validate squash option specified in the resource definition for NFS server

### ISSUE THREAD

**Issue by rohan47** [rook/rook#2037]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
Currently, the Rook NFS operator does't have code to validate the squash option specified in the resource definition.
This feature should add code to validate `none`, `rootId`, `root` and `all` as squash options as mentioned in the [NFS CRD design](https://github.com/rook/rook/blob/master/design/nfs.md#nfs-crd) and should return error if something other than this is mentioned.

**What is use case behind this feature:**
This feature will enable validation for squash option of the NFS server.

**Comment by tsullens** [comment]
Is this still a valid feature request?

**Comment by rohan47** [comment]
@tsullens as per the code, It can take squash option other than `none`, but there is no code that validates the squash option. I am updating the feature request.

**Comment by tsullens** [comment]
@rohan47 that's what I thought after looking through the code, thanks for clarifying and clearing this up.  

Am I right in believing the validation for this should happen in the [NFSController](https://github.com/rook/rook/blob/master/pkg/operator/nfs/controller.go)? If that is the case, I'm wondering what should happen if the squash option is invalid? Defer to the default none? Log an error and discard the export?   

As an extra, it looks to me like `exports.server.accessMode` could also use validation.

**Comment by tsullens** [comment]
I'm really not sure what I was thinking in the previous comment, the feature request pretty clearly points out what we should do for an invalid squash option.  
I have opened a branch which has the changes for this and https://github.com/rook/rook/issues/2036, since it makes sense to validate all of these at once.  
Ignore the first two commit references, I botched the branch for that and just opened a new branch for everything to be clean.  
I guess let me know if there's anything I need to add or to fix - I'd be happy to complete this.
